From da66c21a857c721390792c4e39e2d3ea8ce7a2b7 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 10 May 2022 14:10:26 +0300
Subject: [PATCH 39/64] arm: s32cc: Get A53GPR driver instance based on node
 offset

Get A53GPR driver instance based on node offset instead of
node's name.

Issue: ALB-8823
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/s32cc.dtsi        |  4 ++--
 arch/arm/mach-s32/s32-cc/fdt.c | 14 ++++++++++----
 2 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/arch/arm/dts/s32cc.dtsi b/arch/arm/dts/s32cc.dtsi
index cc1e662a45..7fa4331efa 100644
--- a/arch/arm/dts/s32cc.dtsi
+++ b/arch/arm/dts/s32cc.dtsi
@@ -383,9 +383,9 @@
 		status = "okay";
 	};
 
-	a53_gpr {
+	a53_gpr@4007c400 {
 		compatible = "nxp,s32cc-a53-gpr";
-		reg = <0x0 0x4007C400 0x0 0x100>;
+		reg = <0x0 0x4007c400 0x0 0x100>;
 		status = "okay";
 	};
 
diff --git a/arch/arm/mach-s32/s32-cc/fdt.c b/arch/arm/mach-s32/s32-cc/fdt.c
index 6af7d789b5..1f45454513 100644
--- a/arch/arm/mach-s32/s32-cc/fdt.c
+++ b/arch/arm/mach-s32/s32-cc/fdt.c
@@ -107,13 +107,19 @@ static int get_cores_info(u32 *max_cores_per_cluster,
 
 static bool is_lockstep_enabled(void)
 {
-	int ret;
+	int ret, off;
 	u32 lockstep_enabled = 0;
-	const char *dev_name = "a53_gpr";
 	struct udevice *s32cc_a53_gpr = NULL;
+	const char *a53_compat = "nxp,s32cc-a53-gpr";
 
-	ret = uclass_get_device_by_name(UCLASS_MISC, dev_name,
-					&s32cc_a53_gpr);
+	off = fdt_node_offset_by_compatible(gd->fdt_blob, -1, a53_compat);
+	if (off < 0) {
+		printf("%s: Couldn't find \"%s\" node: %s\n", __func__,
+		       a53_compat, fdt_strerror(off));
+		return false;
+	}
+
+	ret = uclass_get_device_by_of_offset(UCLASS_MISC, off, &s32cc_a53_gpr);
 	if (ret) {
 		printf("%s: No A53 GPR (err = %d)\n", __func__, ret);
 		return false;
-- 
2.17.1

